pkg/http/http.go

```diff
@@ -95,10 +212,23 @@ func NewHTTPClient(cfg ClientConfig, name string) (*http.Client, error) {
 		return nil, err
 	}
 
-	client, err := config_util.NewClientFromConfig(httpClientConfig, name, config_util.WithHTTP2Disabled())
+	// see https://github.com/golang/go/issues/13801
+	if cfg.MaxIdleConns == 0 {
+		cfg.MaxIdleConns = 20000
```

Comment:
I don't get it, why we don't use default? Also `0` means no limit, so 0 has meaning too, we should tolerate such option

Reply:
> I don't get it, why we don't use default? Also `0` means no limit, so 0 has meaning too, we should tolerate such option

Actually, this value is picked from `prometheus http` which we were using before, so thought it would be better to keep it the same. Also yeah 0 does for this is acceptable, 'll fix it.